WooCommerce/Classes/ViewRelated/AppCoordinator.swift

```diff
@@ -49,21 +49,52 @@ final class AppCoordinator {
                 case (true, false):
                     self.validateRoleEligibility {
                         self.displayLoggedInUI()
+                        self.synchronizeAndShowWhatsNew()
                     }
                 }
                 self.isLoggedIn = isLoggedIn
-                self.showWhatsNewIfNeeded()
             }
     }
 }
 
 private extension AppCoordinator {
 
-    /// Displays the What's New Screen.
+    /// Synchronize announcements and present What's New Screen if needed
+    ///
+    func synchronizeAndShowWhatsNew() {
+        stores.dispatch(AnnouncementsAction.synchronizeAnnouncements(onCompletion: { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let announcement):
+                DDLogInfo("📣 Announcements Synced! AppVersion: \(announcement.appVersionName) | AnnouncementVersion: \(announcement.announcementVersion)")
+            case.failure(let error):
+                DDLogInfo("📣 Failed to synchronize announcements: \(error.localizedDescription)")
+            }
+            self.showWhatsNewIfNeeded()
+        }))
+    }
+
+    /// Load saved announcement and display it on What's New component if it was not displayed yet
     ///
     func showWhatsNewIfNeeded() {
-        // TODO: Check the saved Announcement App Version in order to display or not the what's new component
-        stores.dispatch(AnnouncementsAction.synchronizeAnnouncements(onCompletion: { _ in }))
+        stores.dispatch(AnnouncementsAction.loadSavedAnnouncement(onCompletion: { [weak self] result in
+            guard let self = self, let (announcement, displayed) = try? result.get(), !displayed else {
+                return DDLogInfo("📣 There are no announcements to show!")
+            }
+            let whatsNewViewController = WhatsNewFactory.whatsNew(announcement) { [weak self] in
+                self?.tabBarController.dismiss(animated: true)
+            }
+            self.tabBarController.present(whatsNewViewController, animated: true, completion: nil)
+
+            self.stores.dispatch(AnnouncementsAction.markSavedAnnouncementAsDisplayed(onCompletion: { result in
```

Comment:
@Ecarrion Not sire if marking the announcement as displayed right here is the best idea. I'd like to hear from you what you think about doing it here versus moving it to the CTA action triggered by the WhatsNewViewModel.

As far as I can tell, there's only one benefit on leaving this code here:
-  In case user closes the app before pressing the CTA button on what's new component, we would display the announcement once again when the app is reopened. _Doesn't seem a big deal IMHO._

The con is:
- This function is doing several things already.. (bye bye SRP)

Reply:
Ok, cool. I like the idea. I'll be doing this in a few!

Reply:
Done! ✅